apps/vaporgui/MainForm.cpp

```diff
@@ -1060,6 +1074,141 @@ void MainForm::_createDeveloperMenu()
 #endif
 }
 
+void MainForm::createBookmark()
+{
+    bool    ok;
+    QString input = QInputDialog::getText(this, "New Bookmark", "Bookmark Name:", QLineEdit::Normal, "", &ok);
+    if (!ok) return;
+
+    string title = input.toStdString();
+    if (title.empty()) title = "Unnamed Bookmark";
+
+    auto p = GetStateParams();
+
+    p->BeginGroup("Create Bookmark");
+
+    vector<BookmarkParams> bookmarks;
+    for (auto *b : p->GetBookmarks()) bookmarks.push_back(BookmarkParams(*b));
+    p->ClearBookmarks();
+
+    Base16StringStream ss;
+    ss << *_paramsMgr->GetXMLRoot();
+
+    for (auto &b : bookmarks) p->AddBookmark(&b);
+
+    string           activeVizWin = p->GetActiveVizName();
+    ViewpointParams *vpp = _paramsMgr->GetViewpointParams(activeVizWin);
+
+    bool useCustomViewport = vpp->GetValueLong(vpp->UseCustomFramebufferTag, 0);
+    int  customViewportWidth = vpp->GetValueLong(vpp->CustomFramebufferWidthTag, 0);
+    int  customViewportHeight = vpp->GetValueLong(vpp->CustomFramebufferHeightTag, 0);
+
+    int           iconSize = 19;
```

Comment:
What does this encoded string do? Needs some documentation.

Reply:
Done.